### DISCUSSION TITLE
API docs on uploading files are unclear

### DISCUSSION THREAD

**Discussion by aarondfrancis** [vercel/community#225] [Help] [answered]
Hey y'all! I'm trying to create a deployment via API and am running into a few issues. 

The docs on [uploading files](https://vercel.com/docs/rest-api#endpoints/deployments/upload-deployment-files) aren't super clear. They don't mention where the digest should go.

I think it might be a header called `x-now-digest`? Based on this: https://github.com/vercel/vercel/blob/bf3a1fc832f0d131772f1728950c20c9ce2b6275/packages/client/src/upload.ts#L105.

But then if I do that, how do I reference it in the deployment? For the `files` part of a new deployment, the docs say

> A list of objects with the files to be deployed

which isn't totally clear.

So my questions are: 

- do I have to upload the files one by one? Can I upload a ZIP?
- once I upload the files, how to I then reference them to create a deployment?
- when the files are uploaded, they return a blank 200 ok. How long can I reference those before they're gone?
- it looks like the files rate limit is 5k? Is that right? Seems low to me.

Thanks for any help you may be able to provide!

**Comment by aarondfrancis** [comment]
Now I'm wondering if I could upload a single zip file and then unzip it as part of my build step? That might be nice.

**Comment by Cl3arglass** [comment, marked as answer]
Hey @aarondfrancis, we have a support article to share on this topic that includes an example curl request and how to deploy a project with multiple files here: https://vercel.com/support/articles/how-do-i-generate-an-sha-for-uploading-a-file-to-the-vercel-api

At this time, each file needs to be uploaded individually. After upload the files are referenced by SHA when you create a deployment.

**Reply by aarondfrancis** [reply]
Thanks, this article will help. I ended up mucking about with your CLI tool to dump out the requests it was making and just copied those! I appreciate the response.
